GitCommands/Git/GitCommandHelpers.cs

```diff
@@ -417,10 +441,12 @@ public static ArgumentString CleanUpCmd(bool dryRun, bool directories, bool nonI
 
         public static ArgumentString GetAllChangedFilesCmd(bool excludeIgnoredFiles, UntrackedFilesMode untrackedFiles, IgnoreSubmodulesMode ignoreSubmodules = IgnoreSubmodulesMode.None, bool noLocks = false)
         {
-            return new ArgumentBuilder
+            return new GitArgumentBuilder(
+                noLocks && GitVersion.Current.SupportNoOptionalLocks
+                    ? new ArgumentBuilder() { "--no-optional-locks", "status" }.ToString()
```

Comment:
Is `ArgumentBuilder` really necessary here? 
We'd be doing a number of allocations which can be mitigated with a simple `"--no-optional-locks status"`

Reply:
No.  I guess not.